### PR TITLE
*: hide secure information in SHOW PROCESSLIST

### PR DESCRIPTION
**Pull request by tiancaiamao**
`show processlist` should not display something like password.

@zimulala @coocood @shenli

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ type Node interface {
 	Text() string
 	// SetText sets original text to the Node.
 	SetText(text string)
+	// SecureText is different from Text that it hide password information.
+	SecureText() string
```
Comment (coocood) [inline ast/ast.go]: I think this method should not be put in Node interface.
How about defining a CrusialStmt interface which only has a SecureText method?

Comment (tiancaiamao) [reply ast/ast.go]: So we implement `SecureText()` for so many

```
*ast.CreateUserStmt, *ast.DropUserStmt, *ast.AlterUserStmt, *ast.SetPwdStmt, *ast.GrantStmt,
*ast.RevokeStmt, *ast.AlterTableStmt, *ast.CreateDatabaseStmt, *ast.CreateIndexStmt, *ast.CreateTableStmt,
*ast.DropDatabaseStmt, *ast.DropIndexStmt, *ast.DropTableStmt, *ast.RenameTableStmt, *ast.TruncateTableStmt:
```

and most of those implemention is just ... ?

```
func (x *XXX) SecureText() string {
    return x.Text()
}
```

Comment (coocood) [reply ast/ast.go]: Well, not all crucial statement need to have secure text.
Then how about define a `SensitiveStatement` which has `SecureText`?

Comment (coocood) [reply ast/ast.go]:
```
if ss, ok := node.(SensitiveStmt) {
    text = ss.SecureText()
} else {
    text = node.Text()
}
```

Comment (shenli) [reply ast/ast.go]: Could we add SecureText to StmtNode interface?

Comment (coocood) [reply ast/ast.go]: @shenli 
`SecureText` only needed by 4 statements, I think adding it to all StmtNode is not worthy.

Comment (shenli) [reply ast/ast.go]: ok

**Comment by shenli** [comment]
@tiancaiamao Any update?

**Comment by tiancaiamao** [comment]
PTAL @coocood @shenli

File: ast/misc.go [hunk 2]
```diff
@@ -406,6 +408,11 @@ type SetPwdStmt struct {
 	Password string
 }
 
+// SecureString implements SensitiveStatement interface.
+func (n *SetPwdStmt) SecureString() string {
```
Comment (coocood) [inline ast/misc.go]: SecureText or SecureString?

**Comment by coocood** [comment]
Please add test.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @zimulala @shenli

**Comment by zimulala** [comment]
LGTM

**Pull request merged**